### PR TITLE
Allow random menu images for subgames

### PR DESCRIPTION
**Pull request by sfan5**
#2911 but with suggestions by @est31 and @kilbith implemented.

### CODE REVIEW

File: builtin/common/misc_helpers.lua [hunk 1]
```diff
@@ -184,9 +185,22 @@ function string.split(str, delim, include_empty, max_splits, sep_is_pattern)
 end
 
 --------------------------------------------------------------------------------
+function table.indexof(list, val)
+	for i = 1, #list do
+		if list[i] == val then
+			return i
+		end
+	end
+	return -1
```
Comment (est31) [inline builtin/common/misc_helpers.lua]: Isn't it lua tradition to return 0 in these cases?

**Comment by est31** [comment]
:+1:

File: builtin/common/misc_helpers.lua (hunk 1 above)
Comment (sfan5) [inline builtin/common/misc_helpers.lua]: No idea, I just used -1 because thats what javascript does too.

**Comment by Amaz1** [comment]
Thanks sfan5!

**Comment by nerzhul** [comment]
:+1:

**Comment by kilbith** [comment]
I just thought : this should have a documentation on how naming the files for using them randomly.

**Comment by est31** [comment]
899491325938d0b1e5bf701d33d24bb4283affaa

**Comment by est31** [comment]
@kilbith see 403e6e6c9cf6be5e16e3f1f9bd7805fe9b3006cd

**Comment by kilbith** [comment]
Thnaks.

**Pull request closed without merging**